File: hammer-gui/src/main/java/dev/vini2003/hammer/gui/debug/common/screen/handler/DebugScreenHandler.java
package dev.vini2003.hammer.gui.debug.common.screen.handler;

import dev.vini2003.hammer.gui.api.common.screen.handler.BaseScreenHandler;
import dev.vini2003.hammer.gui.api.common.widget.arrow.ArrowWidget;
import dev.vini2003.hammer.gui.api.common.widget.bar.BarWidget;
import dev.vini2003.hammer.gui.api.common.widget.bar.FluidBarWidget;
import dev.vini2003.hammer.gui.api.common.widget.bar.ImageBarWidget;
import dev.vini2003.hammer.gui.api.common.widget.bar.SpriteBarWidget;
import dev.vini2003.hammer.gui.api.common.widget.button.ButtonWidget;
import dev.vini2003.hammer.gui.api.common.widget.image.ImageWidget;
import dev.vini2003.hammer.gui.api.common.widget.item.ItemStackWidget;
import dev.vini2003.hammer.gui.api.common.widget.item.ItemWidget;
import dev.vini2003.hammer.gui.api.common.widget.list.ListWidget;
import dev.vini2003.hammer.gui.api.common.widget.list.slot.SlotListWidget;
import dev.vini2003.hammer.gui.api.common.widget.panel.PanelWidget;
import dev.vini2003.hammer.gui.api.common.widget.screen.ScreenWidget;
import dev.vini2003.hammer.gui.api.common.widget.slot.SlotWidget;
import dev.vini2003.hammer.gui.api.common.widget.tab.TabWidget;
import dev.vini2003.hammer.gui.api.common.widget.text.TextAreaWidget;
import dev.vini2003.hammer.gui.api.common.widget.text.TextEditorWidget;
import dev.vini2003.hammer.gui.api.common.widget.text.TextFieldWidget;
import dev.vini2003.hammer.gui.api.common.widget.text.TextWidget;
import dev.vini2003.hammer.gui.api.common.widget.toggle.ToggleWidget;
import dev.vini2003.hammer.gui.debug.common.util.WidgetUtil;
import dev.vini2003.hammer.gui.registry.common.debug.HGUIScreenHandlerTypes;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.ItemStack;
import net.minecraft.screen.ScreenHandlerType;
import org.jetbrains.annotations.Nullable;

import java.awt.*;

public class DebugScreenHandler extends BaseScreenHandler {
	public DebugScreenHandler(int syncId, PlayerEntity player) {
		super(HGUIScreenHandlerTypes.DEBUG, syncId, player);
	}
	
	@Override
	public void init(int width, int height) {
		var all = WidgetUtil.createAll();
		
		var screen = new PanelWidget();
		screen.setWidth(width);
		screen.setHeight(height);
		
		add(screen);

		var x = 18.0F;
		var y = 18.0F;
		
		var i = 0;
		
		for (var widget : all) {
			i++;
			
			widget.setX(x);
			widget.setY(y);
			
			x += widget.getWidth() + 18.0F;
			
			if (x >= width) {
				x = 18.0F;
				var maxH = widget.getHeight();
				// find highest widget in this x and add to y
				for (var w : all) {
					if (w.getY() == y && w.getHeight() > maxH) {
						maxH = w.getHeight();
					}
				}
				y += maxH + 18.0F;
			}
			
			screen.add(widget);
		}
	}
	
	@Override
	public boolean canUse(PlayerEntity player) {
		return true;
	}
}

File: src/main/kotlin/com/github/vini2003/blade/testing/DebugScreenHandler.kt
package com.github.vini2003.blade.testing

import com.github.vini2003.blade.client.utilities.Instances
import com.github.vini2003.blade.common.data.Position
import com.github.vini2003.blade.common.data.Size
import com.github.vini2003.blade.common.handler.BaseScreenHandler
import com.github.vini2003.blade.common.widget.base.ButtonWidget
import com.github.vini2003.blade.common.widget.base.SlotWidget
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.screen.ScreenHandlerType

class DebugScreenHandler(syncId: Int, player: PlayerEntity) : BaseScreenHandler(DebugContainers.DEBUG_HANDLER, syncId, player) {
    init {
        addInventory(0, player.inventory)

        val slot: SlotWidget = SlotWidget(0, 0)

        slot.setPosition(Position({16F}, {16F}))
        slot.setSize(Size({18F}, {18F}))

        val button: ButtonWidget = ButtonWidget {
            if (it.focused) print("I was clicked!")
        }

        button.setPosition(Position({32F}, {16F}))
        button.setSize(Size({80F}, {16F}))

        addWidget(slot)

        addWidget(button)
    }
}